setuptools/dist.py

```diff
@@ -25,6 +25,14 @@
 from .py36compat import Distribution_parse_config_files
 
 
+skip_install_eggs = False
```

Comment:
This variable appears to be undocumented and untested. It could be optimized away.

I wonder if instead of setting a global flag that tweaks the inner behavior of the Distribution object for all users if it wouldn't be better for the caller that would use this functionality to instead create a:

```
class Distribution(setuptools.dist.Distribution):
    def fetch_build_eggs(self, eggs):
        raise SetupRequirementsError(...)
```

It's not yet clear to me where a user or application would hook into this behavior.

Also, the name 'skip_install_eggs' doesn't match the behavior. It's 'fail on install eggs'.

Reply:
This variable is for internal use only. The public api is `get_requires_for_build_wheel`